openshift/topics/introduction.adoc

```diff
@@ -1,35 +1,66 @@
+
+=== Comparison: {project_openshift_product_name} Image versus Red Hat Single Sign-On
+The {project_openshift_product_name} image version number {project_version} is based on {project_name} {project_version}. There are some differences in functionality between the {project_openshift_product_name} image and {project_name}:
+
+The {project_openshift_product_name} image includes all of the functionality of {project_name}. In addition, the {project_name}-enabled JBoss EAP image automatically handles OpenID Connect or SAML client registration and configuration for *_.war_* deployments that contain *<auth-method>KEYCLOAK</auth-method>* or *<auth-method>KEYCLOAK-SAML</auth-method>* in their respective *web.xml* files.
 
 [[sso-templates]]
-Red Hat offers multiple OpenShift application templates utilizing the {project_openshift_product_name} image version number {project_version}. These define the resources needed to develop {project_name} {project_version} server based deployment and can be split into the following two categories:
+=== Templates for use with this software
+
+Red Hat offers multiple OpenShift application templates using the {project_openshift_product_name} image version number {project_version}. These templates define the resources needed to develop {project_name} {project_version} server based deployment. The templates can mainly be split into two categories: passthrough templates and reencryption templates. Some other miscellaneous templates also exist.
 
 [[passthrough-templates]]
-* Templates using HTTPS and JGroups keystores and a truststore for the {project_name} server, all prepared beforehand. These secure the TLS communication using link:{ocp311docs_passthrough_route_link}[passthrough TLS termination]:
+==== Passthrough templates
+
+These templates use HTTPS and JGroups keystores and a truststore for the {project_name} server.  They secure the TLS communication using passthrough TLS termination.
 
-** *_{project_templates_version}-https_*: {project_name} {project_version} backed by internal H2 database on the same pod.
-** *_{project_templates_version}-postgresql_*: {project_name} {project_version} backed by ephemeral PostgreSQL database on a separate pod.
-** *_{project_templates_version}-postgresql-persistent_*: {project_name} {project_version} backed by persistent PostgreSQL database on a separate pod.
+* _{project_templates_version}-https_*: {project_name} {project_version} backed by internal H2 database on the same pod.
+
+* _{project_templates_version}-postgresql_*: {project_name} {project_version} backed by ephemeral PostgreSQL database on a separate pod.
+
+* _{project_templates_version}-postgresql-persistent_*: {project_name} {project_version} backed by persistent PostgreSQL database on a separate pod.
 
 [NOTE]
 Templates for using {project_name} with MySQL / MariaDB databases have been removed and are not available since {project_name} version 7.4.
 
+==== Reencryption templates
```

Comment:
Maybe prefer `Re-encryption templates` (IOW form with the dash between the words) over the `Reencryption templates` above? (upstream OpenShift docs uses `re-encrypt` form more often too:
  * https://docs.openshift.com/container-platform/4.1/networking/routes/secured-routes.html#nw-ingress-creating-a-reencrypt-route-with-a-custom-certificate_secured-routes)